packages/design-tokens/design-token-previews/utils.tsx

```diff
@@ -0,0 +1,117 @@
+import iconSizeTokens from "../src/tokens/size.tokens.json";
+import spaceTokens from "../src/tokens/space.tokens.json";
+import borderStyleTokens from "../src/tokens/border-style.tokens.json";
+import borderWidthTokens from "../src/tokens/border-width.tokens.json";
+import borderRadiiTokens from "../src/tokens/border-radius.tokens.json";
+
+import {
+  createBorderRadiiPreview,
+  createSpacePreview,
+  createColorPreview,
+  createFontSizePreview,
+  createIconSizePreview,
+  createFontWeightPreview,
+  createBorderStylePreview,
+  createBorderWidthPreview,
+} from "./utils/previews";
+
+import { hexToHsla, hexToRgb } from "./utils/colors";
+import { Token, TokenEntry } from "./types/Token";
+import { Column, Row } from "./types/TokenTable";
+
+import { CopyToClipboardButton } from "./CopyToClipboardButton";
+
+const getKey = (object: Record<string, unknown>): string => {
+  return keys(object)[0];
+};
+
+const getValue = ([, token]: TokenEntry): string => {
+  return token.value;
+};
+
+const getComment = ([, token]: TokenEntry): string => {
+  return token.comment || "";
+};
+
+const formatTokenName = (prefix: string, suffix: string) => {
+  const tokenName = camelCase(`${prefix}${capitalize(suffix)}`);
+  return (
+    <div style={{ display: "flex" }}>
+      <div>{tokenName}</div>
+      <div>
+        <CopyToClipboardButton textToCopy={tokenName} />
+      </div>
+    </div>
+  );
+};
+
+export const TOKENS = {
+  [getKey(borderRadiiTokens)]: [
+    { name: "Pixels", transform: getValue },
+    { name: "Preview", transform: createBorderRadiiPreview },
+  ],
+  [getKey(borderWidthTokens)]: [
+    { name: "Pixels", transform: getValue },
+    { name: "Preview", transform: createBorderWidthPreview },
+  ],
+  [getKey(borderStyleTokens)]: [
+    { name: "Style", transform: getValue },
+    { name: "Preview", transform: createBorderStylePreview },
+  ],
+  [getKey(spaceTokens)]: [
+    { name: "Pixels", transform: getComment },
+    { name: "Rems", transform: getValue },
+    { name: "Preview", transform: createSpacePreview },
+  ],
+  [getKey(iconSizeTokens)]: [
+    { name: "Pixels", transform: getComment },
+    { name: "Rems", transform: getValue },
+    { name: "Preview", transform: createIconSizePreview },
+  ],
+  [getKey(fontWeightTokens)]: [
+    { name: "Weight", transform: getValue },
+    { name: "Preview", transform: createFontWeightPreview },
+  ],
+  [getKey(fontSizeTokens)]: [
+    { name: "Pixels", transform: getComment },
+    { name: "Rems", transform: getValue },
+    { name: "Preview", transform: createFontSizePreview },
+  ],
+  [getKey(colorTokens)]: [
+    { name: "Hex", transform: getValue },
+    {
+      name: "RGB",
+      transform: ([, token]: TokenEntry): string => hexToRgb(token.value),
+    },
+    {
+      name: "Hsla",
+      transform: ([, token]: TokenEntry): string => hexToHsla(token.value),
+    },
+    { name: "Preview", transform: createColorPreview },
+  ],
+};
+
+export const renderRows = (
```

Comment:
I would suggest giving this function a different name because it isn't rendering but parsing the token data. The render happens on the `TokensTable` component (after iterating on this function result).